packages/core/contracts/financial-templates/contract-for-difference/ContractForDifference.sol

```diff
@@ -147,8 +149,8 @@ contract ContractForDifference is Testable, Lockable {
 
         collateralToken.safeTransferFrom(msg.sender, address(this), collateralUsed);
 
-        require(longToken.mint(msg.sender, tokensToCreate));
-        require(shortToken.mint(msg.sender, tokensToCreate));
+        require(longToken.mint(msg.sender, tokensToCreate), "Minting long failed");
+        require(shortToken.mint(msg.sender, tokensToCreate), "Minting short failed");
```

Comment:
Do we know if require messages add gas to the _execution_? If so, I'd be in favor of dropping these given that they are _so_ unlikely to happen, and we should be able to identify the issue via a trace if it happened on mainnet.

They are unlikely because:
- The ERC20 contract we're using reverts on failure and always returns true. So we'd need to pass in a different ERC20.
- Assuming the contract has the right permissions, minting failure should be nearly impossible.


I could even see us dropping the require entirely due to the fact that we know the implementation of the ERC20 being passed in (and that it always returns true).

Thoughts?

Reply:
So I tested this out in transaction execution:
- With requires + messages `gasUsed` was `170563`
- With requires and no messages `gasUsed` was `170563`
- no requires at all `gasUsed` was `170539`

So net difference of 24 gas. a fraction of a fraction of the total transaction. I agree with you though that we can remove them entirely as they will revert upstream.

I've done this through and validates it works just the same via the existing unit tests.